examples/TestAsync.purs

```diff
@@ -0,0 +1,48 @@
+module Main where
+
+import Prelude
+import Data.Either
+import Data.Maybe
+import Control.Monad.Eff
+import Control.Monad.Eff.Console(log,CONSOLE())
+import Control.Monad.Eff.Exception
+
+import Node.FS
+import qualified Node.FS.Async as A
+import qualified Node.Path as FP
+import qualified Node.Buffer as B 
+
+-- exercise the file descriptor based async IO functions
+
+main :: forall eff . Eff (fs::FS,err::EXCEPTION,console::CONSOLE|eff) Unit
+main = do
+  let path1 = FP.concat( ["examples", "TestAsync.purs"] )
+      path2 = FP.concat( ["examples", "TestAsync.purs.partial"] )
+      buf = B.create 1000
+  A.fdOpen path1 R Nothing $ \v -> case v of
+    (Left err) -> log ("err:" ++ show err)
```

Comment:
This pattern of matching on the `Left` constructor every time should be abstracted away using a separate function. Also maybe throw an exception rather than just logging that an error happened? Then we could put this file into `test/` and use it as a test.